src/server/game/Entities/Player/Player.cpp

```diff
@@ -18742,9 +18742,34 @@ bool Player::LoadFromDB(ObjectGuid guid, CharacterDatabaseQueryHolder* holder)
 
     m_achievementMgr->CheckAllAchievementCriteria(this);
     m_questObjectiveCriteriaMgr->CheckAllQuestObjectiveCriteria(this);
+
+    _PushQuests();
     return true;
 }
 
+void Player::_PushQuests()
+{
+    ObjectMgr::QuestMap const& questMap = sObjectMgr->GetQuestTemplatesAutoPush();
+
+    for (ObjectMgr::QuestMap::const_iterator it = questMap.cbegin(); it != questMap.cend(); ++it)
+    {
+        QuestStatusMap::const_iterator result = m_QuestStatus.find(it->first);
```

Comment:
All of the checks (including quest status) should be replaced with just a simple function call `CanTakeQuest(it->second, false)`